infra/lambda/datasets-endpoint/function.py

```diff
@@ -0,0 +1,482 @@
+"""
+Dataset Lambda handler function.
+"""
+
+import uuid
+from datetime import datetime
+
+import boto3
+from dateutil.tz import gettz
+from jsonschema import ValidationError, validate
+
+DYNAMODB = boto3.client("dynamodb")
+
+DS_PRIMARY_KEYS = ("id", "type")
+DS_ATTRIBUTES = ("title", "owning_group")
+DS_ATTRIBUTES_EXT = ("created_at",)
+
+REQUEST_SCHEMA = {
+    "type": "object",
+    "properties": {
+        "httpMethod": {"type": "string", "enum": ["GET", "POST", "PATCH", "DELETE"]},
+        "body": {"type": "object"},
+    },
+    "required": ["httpMethod", "body"],
+}
+
+# TODO: implement GET response paging
+# TODO: allow Dataset delete only if no Dataset Version exists
+# TODO: don't assume that all Dataset attributes are strings ("S")
+
+
+# fmt: off
+def lambda_handler(event, context):  # pylint:disable=unused-argument,inconsistent-return-statements,too-many-return-statements
+    """Main Lambda entry point."""
+# fmt: on
+
+    # request validation
+    try:
+        validate(event, REQUEST_SCHEMA)
+        method = event["httpMethod"]
+    except ValidationError as err:
+        return {"statusCode": 400, "body": {"message": f"Bad Request: {err.message}."}}
+
+    if method == "POST":
+        return post_method(event)
+
+    if method == "GET" and "id" in event["body"] and "type" in event["body"]:
+        return get_method_single(event)
+
+    if method == "GET" and ("title" in event["body"] or "owning_group" in event["body"]):
+        return get_method_filter(event)
+
+    if method == "GET" and event["body"] == {}:
+        return get_method_all()
```

Comment:
If you mean having nested condition under `if method == "GET"` condition then I agree. I was considering this as well.

Reply:
Done in 54e6750c180b8af654e846f0a74a5628be0786cb

Reply:
Actually, I read your comment once again and now understand what you mean. `get_method_filter` and `get_method_all` are separate functions because in DynoDB they use different approach to get data - `filter` is using `query` by index and `all` is doing full table `scan`.